spec/index.bs

```diff
@@ -646,15 +635,46 @@ interface GPURenderPassEncoder : GPUProgrammablePassEncoder {
     // Base instance must be set to zero inside the indirect buffer
     void drawIndirect(GPUBuffer indirectBuffer, u64 indirectOffset);
     void drawIndexedIndirect(GPUBuffer indirectBuffer, u64 indirectOffset);
+};
 
-    // TODO add missing commands
+interface GPURenderPassEncoder : GPURenderEncoderBase {
+    // The default viewport is (0.0, 0.0, w, h, 0.0, 1.0), where w and h are the dimensions of back buffer
+    void setViewport(float x, float y, float width, float height, float minDepth, float maxDepth);
+
+    // The default scissor rectangle is (0, 0, w, h), where w and h are the dimensions of back buffer.
+    // Width and height must be greater than 0. Otherwise, an error will be generated.
+    void setScissorRect(u32 x, u32 y, u32 width, u32 height);
+
+    void setBlendColor(GPUColor color);
+    void setStencilReference(u32 reference);
+
+    void executeBundles(sequence<GPURenderBundle> bundles);
+    void endPass();
+};
+
+dictionary GPURenderBundleDescriptor : GPUObjectDescriptorBase {
+};
+
+interface GPURenderBundle : GPUObjectBase {
+};
+
+interface GPURenderBundleEncoder : GPURenderPassEncoderBase {
```

Comment:
`GPURenderPassEncoderBase` does not exist. I'd suggest using `GPUObjectBase` for now instead of defining yet another empty interface such as `GPURenderBundleEncoderBase`.

```suggestion
interface GPURenderBundleEncoder : GPUObjectBase {
```